cpp/src/io/parquet/page_enc.cu

```diff
@@ -2141,6 +2164,304 @@ CUDF_KERNEL void __launch_bounds__(block_size, 8)
+    auto const in_range   = t_idx < s->page.num_valid;
+    auto const val        = in_range ? prefix_lengths[t_idx] : 0;
+    packer.add_value(val, in_range);
+  }
+
+  auto const suffix_ptr = packer.flush();
+  __syncthreads();
+
+  // encode suffix lengths
+  if (t == 0) {
+    packer.init(
+      suffix_ptr, s->page.num_valid, reinterpret_cast<int32_t*>(delta_shared), &temp_storage);
+  }
+  __syncthreads();
+
+  size_t non_zero     = 0;
+  size_t suffix_bytes = 0;
+
+  for (int idx = 0; idx < s->page.num_valid; idx += block_size) {
+    size_type const t_idx = idx + t;
+    auto const in_range   = t_idx < s->page.num_valid;
+    int32_t val           = 0;
+    if (in_range) {
+      size_type const leaf_idx = has_leaf_nulls ? offsets_map[t_idx] : t_idx;
+      auto const byte_arr      = get_byte_array(leaf_idx + s->page_start_val);
+      val                      = byte_arr.length - prefix_lengths[t_idx];
+      if (val > 0) {
+        non_zero++;
+        suffix_bytes += val;
+      }
+    }
+    packer.add_value(val, in_range);
+  }
+
+  auto const strings_ptr = packer.flush();
+
+  non_zero = block_reduce(temp_storage.reduce_storage).Sum(non_zero);
+  __syncthreads();
+  suffix_bytes = block_reduce(temp_storage.reduce_storage).Sum(suffix_bytes);
+  if (t == 0) { avg_suffix_len = util::div_rounding_up_unsafe(suffix_bytes, non_zero); }
+  __syncthreads();
+
+  // Now copy the byte array data. For shorter suffixes (<= 64 bytes), it is faster to use
+  // memcpy on a string-per-thread basis. For longer suffixes, it's better to use a parallel
+  // approach.
+  // Note: this param still needs tuning.
+  constexpr size_t suffix_cutoff = 64;
+
+  size_t str_data_len = 0;
+  if (avg_suffix_len <= suffix_cutoff) {
+    for (int idx = 0; idx < s->page.num_valid; idx += block_size) {
+      size_type const t_idx = idx + t;
+      size_type s_len = 0, pref_len = 0, suff_len = 0;
+      uint8_t const* s_ptr = nullptr;
+      if (t_idx < s->page.num_valid) {
+        size_type const leaf_idx = has_leaf_nulls ? offsets_map[t_idx] : t_idx;
+        auto const byte_arr      = get_byte_array(leaf_idx + s->page_start_val);
+        s_len                    = byte_arr.length;
+        s_ptr                    = byte_arr.data;
+        pref_len                 = prefix_lengths[t_idx];
+        suff_len                 = byte_arr.length - pref_len;
+      }
+
+      // calculate offsets into output
+      size_type s_off, total;
+      block_scan(temp_storage.scan_storage)
+        .ExclusiveScan(suff_len, s_off, str_data_len, cub::Sum(), total);
+
+      if (t_idx < s->page.num_valid) {
+        auto const dst = strings_ptr + s_off;
+        memcpy(dst, s_ptr + pref_len, suff_len);
+      }
+      str_data_len += total;
+      __syncthreads();
+    }
+  } else {
+    int t0 = 0;  // thread 0 for each string
+    for (int idx = 0; idx < s->page.num_valid; idx++) {
+      // calculate ids for this string
+      int mytid = t - t0;
```

Comment:
maybe just tid? the prefix does not do much, IMO.

Reply:
Just trying to convey that it's a new thread id.  I guess `tid` works for that purpose.